scripts/train/train.py

```diff
@@ -40,29 +44,100 @@
 log = logging.getLogger(__name__)
 
 
-def validate_config(cfg: DictConfig):
+@dataclass
+class TrainConfig:
+    model: Dict[str, Any] = MISSING
+    tokenizer: Dict[str, Any] = MISSING
+    optimizer: Dict[str, Any] = MISSING
+    scheduler: Dict[str, Any] = MISSING
+    train_loader: Dict[str, Any] = MISSING
+    device_train_batch_size: int = MISSING
+    device_eval_batch_size: int = MISSING
+    max_duration: Union[int, str] = MISSING
+    eval_interval: Union[int, str] = MISSING
+    precision: str = MISSING
+    max_seq_len: int = MISSING
+    seed: int = MISSING
+
+    code_paths: Optional[List[str]] = None
```

Comment:
Can we organize these and below? maybe just separate them via new lines and comments like Mandatory model training configs, Optional fsdp data, fine-tuning, and eval configs, etc.